test/test_content_type.py

```diff
@@ -0,0 +1,220 @@
+                        'text': '<foo>',
+                        'content-type': 'text/html'
+                    },
+                    'escaped': None,
+                },
+                'NotFoundError': {
+                    'unescaped': {
+                        'text': '{"Code":"NotFoundError","Message":"<foo>"}',
+                        'content-type': 'application/json'
+                    },
+                    'escaped': None,
+                },
+                'Exception': {
+                    # Chalice's `_unhandled_exception_to_response` returns a
+                    # stacktrace if debug is enabled
+                    'unescaped': {
+                        'text': (
+                            'Traceback (most recent call last):\n'
+                            '...\n'
+                            'Exception: <foo>'
+                        ),
+                        'content-type': 'text/plain'
+                    } if config.debug else {
+                        'text': (
+                            '{"Code":"InternalServerError",'
+                            '"Message":"An internal server error occurred."}'
+                        ),
+                        'content-type': 'application/json'
+                    },
+                    'escaped': {
+                        'text': (
+                            '<html><body>'
+                            '<h1>Status 500</h1>'
+                            '<pre>"Traceback (most recent call last):\\n'
+                            '...\\n'
+                            'Exception: &lt;foo&gt;"</pre>'
+                            '</body></html>'
+                        ),
+                        'content-type': 'text/html'
+                    } if config.debug else {
+                        'text': (
+                            '{"Code":"InternalServerError",'
+                            '"Message":"An internal server error occurred."}'
+                        ),
+                        'content-type': 'application/json'
+                    }
+                },
+                "ChaliceUnhandledError": {
+                    'unescaped': {
+                        'text': (
+                            'Traceback (most recent call last):\n'
+                            '...\n'
+                            'chalice.app.ChaliceUnhandledError: <foo>'
+                        ),
+                        'content-type': 'text/plain'
+                    } if config.debug else {
+                        'text': '{"Code":"InternalServerError",'
+                                '"Message":"An internal server error occurred."}',
+                        'content-type': 'application/json'
+                    },
+                    'escaped': None,
+                }
+            }
+
+        for debug in (0, 1):
+            with mock.patch.object(type(config), 'debug', debug):
+                with mock.patch.object(Chalice, 'debug', debug):
```

Comment:
```suggestion
            with mock.patch.object(type(config), 'debug', debug):
                with mock.patch.object(Chalice, 'debug', debug):
```

Why do you need to patch both? Isn't Chalice.debug derived from azul.debug?

Reply:
Patching Azul's `config.debug` alone isn't enough since the `debug` in Chalice isn't dynamic after being set as part of the AzulChaliceApp constructor.
```
super().__init__(app_name, debug=config.debug > 0, configure_logs=False)
```

Reply:
Then why do you need to patch `config.debug`?

Reply:
The patch of `config.debug` was done so that `get_test_values()` could use the value to know which expected values to return.

I've removed the patching of `config.debug`, and `get_test_values()` now looks at `self.app_module.app.debug` instead.